yetanotherco/aligned_layer
# feat: whitelist

Most important changes are on: https://github.com/yetanotherco/eigenlayer-middleware/tree/feat-whitelist

This PR adds a whitelist to registry coordinator, that only allows whitelisted operators to register with aligned. This will be used in holesky.

# How to test
1. `make deps`
2. `make anvil_deploy_aligned_contracts` (the current anvil state has new contract deployed but to test deployment works)
3. `make operator_full_registration` (this will whitelist operator)
4. `make operator_register_with_eigen_layer operator_mint_mock_tokens operator_deposit_into_mock_strategy operator_register_with_aligned_layer CONFIG_FILE=./config-files/config-operator-2.yaml` (It should fail with `execution reverted: revert: Caller is not whitelisted`)
5. You can test trying to whitelist an operator from another account by running 
```
cast send \
  --rpc-url="http://localhost:8545" \
  --private-key=0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6 \
  0x84eA74d481Ee0A5332c457a4d796187F6Ba67fEB 'add(address)' \
  0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc
```
It should fail with `execution reverted: revert: Ownable: caller is not the owner`
